symphony-bdk-core/src/test/java/com/symphony/bdk/core/service/message/util/MessageParserTest.java

```diff
@@ -0,0 +1,134 @@
+    assertEquals(1, cashtags.size());
+    assertEquals("hello", cashtags.get(0));
+  }
+
+  @Test
+  public void testGetEmojis() throws MessageParserException {
+    Map<String, String> emojisMap = MessageParser.getEmojis(message);
+    assertEquals(1, emojisMap.size());
+    assertEquals("{grinning=\uD83D\uDE00}", emojisMap.toString());
+  }
+
+  @Test
+  public void testEmptyData() throws MessageParserException {
+    when(message.getData()).thenReturn("");
+    assertTrue(MessageParser.getCashtags(message).isEmpty());
+    assertTrue(MessageParser.getEmojis(message).isEmpty());
+  }
+
+  @Test
+  public void testNullData() throws MessageParserException {
+    when(message.getData()).thenReturn(null);
+    assertTrue(MessageParser.getMentions(message).isEmpty());
+    assertTrue(MessageParser.getEmojis(message).isEmpty());
+  }
+
+  @Test
+  public void testJsonProcessingException() {
+    when(message.getData()).thenReturn("Unparsable json");
+    try {
+      MessageParser.getMentions(message);
+      fail("Should have throw a MessageParserException");
+    } catch (Exception e) {
+      assertTrue(e instanceof MessageParserException);
+    }
+  }
+
+  private String getEntityData() {
+    return "{\n"
+        + "  \"0\": {\n"
+        + "    \"id\": [\n"
+        + "      {\n"
+        + "        \"type\": \"com.symphony.user.userId\",\n"
+        + "        \"value\": \"13056700580915\"\n"
+        + "      }\n"
+        + "    ],\n"
+        + "    \"type\": \"com.symphony.user.mention\"\n"
+        + "  },\n"
+        + "  \"1\": {\n"
+        + "    \"id\": [\n"
+        + "      {\n"
+        + "        \"type\": \"org.symphonyoss.taxonomy.hashtag\",\n"
+        + "        \"value\": \"bot\"\n"
+        + "      }\n"
+        + "    ],\n"
+        + "    \"type\": \"org.symphonyoss.taxonomy\",\n"
+        + "    \"version\": \"1.0\"\n"
+        + "  },\n"
+        + "  \"2\": {\n"
+        + "    \"id\": [\n"
+        + "      {\n"
+        + "        \"type\": \"org.symphonyoss.fin.security.id.ticker\",\n"
+        + "        \"value\": \"hello\"\n"
+        + "      }\n"
+        + "    ],\n"
+        + "    \"type\": \"org.symphonyoss.fin.security\",\n"
+        + "    \"version\": \"1.0\"\n"
+        + "  },\n"
+        + "  \"3\": {\n"
+        + "    \"type\": \"com.symphony.emoji\",\n"
+        + "    \"version\": \"1.0\",\n"
+        + "    \"data\": {\n"
+        + "      \"annotation\": \"grinning\",\n"
+        + "      \"family\": \"\",\n"
+        + "      \"size\": \"normal\",\n"
+        + "      \"unicode\": \"\uD83D\uDE00\"\n"
+        + "    }\n"
+        + "  },\n"
+        + "  \"4\": {\n"
+        + "    \"id\": [\n"
+        + "      {\n"
+        + "        \"type\": \"com.symphony.user.userId\",\n"
+        + "        \"value\": \"1305690252351\"\n"
+        + "      }\n"
+        + "    ],\n"
+        + "    \"type\": \"com.symphony.user.mention\"\n"
+        + "  }\n"
+        + "}";
```

Comment:
Would be better to have a JSON file in https://github.com/SymphonyPlatformSolutions/symphony-api-client-java/tree/master/symphony-bdk-core/src/test/resources/message

Reply:
sure I will add it there!